### ISSUE TITLE
upgrade steps for local_o365 plugin for 3.9 branch broken.

### ISSUE THREAD

**Issue by danmarsden** [microsoft/o365-moodle#1985]
https://github.com/microsoft/o365-moodle/blob/MOODLE_39_STABLE/local/o365/db/upgrade.php#L657

https://github.com/microsoft/o365-moodle/blob/MOODLE_39_STABLE/local/o365/db/upgrade.php#L738

$result is undefined, so will always return false and these upgrade steps are not performed.

Only affects the 3.9 branch - $result isn't used in the other branches.

**Comment by danmarsden** [comment]
stuff like: 
```
if ($result && $oldversion < 2020071506) {
```
should be:
```
if ($oldversion < 2020071506) {

**Comment by weilai-irl** [comment]
Hi @danmarsden,

Thank you for reporting this. We have fixed this in the latest release 3.9.9.
I'll close the issue.

Regards,
Lai
